src/main/resources/xslt-files/EMD_doi_datacite_v4.1.xsl

```diff
@@ -779,13 +779,50 @@
     <!-- emd:rights/dct:accessRights to datacite rights -->
     <!-- ==================================================== -->
     <xsl:template match="emd:rights">
-        <xsl:apply-templates select="dcterms:accessRights"/>
-
+        <xsl:variable name="access-rights" select="dcterms:accessRights"/>
         <xsl:variable name="licenses" select="dcterms:license"/>
+        <xsl:variable name="doi" select="//emd:identifier/dc:identifier[@eas:scheme='DOI']/text()"/>
+        <xsl:variable name="origin-doi" select="if (starts-with($doi, '10.17026/dans-')) then 'DANS' else 'OTHER'"/>
+        <xsl:variable name="access-rights-datacite">
+            <xsl:choose>
+                <xsl:when test="$access-rights ='NO_ACCESS' and $origin-doi != 'DANS'">
+                    <xsl:value-of select="'OPEN_ACCESS'"/>
+                </xsl:when>
+                <xsl:otherwise>
+                    <xsl:value-of select="$access-rights"/>
+                </xsl:otherwise>
+            </xsl:choose>
+        </xsl:variable>
+
+        <xsl:if test="$access-rights">
+            <xsl:element name="rights">
+                <xsl:choose>
+                    <xsl:when test="$access-rights-datacite = 'OPEN_ACCESS_FOR_REGISTERED_USERS'">
+                        <xsl:value-of select="'info:eu-repo/semantics/openAccess'"/>
+                    </xsl:when>
+                    <xsl:when test="$access-rights-datacite = 'OPEN_ACCESS'">
+                        <xsl:value-of select="'info:eu-repo/semantics/openAccess'"/>
+                    </xsl:when>
+                    <xsl:when test="$access-rights-datacite = 'GROUP_ACCESS'">
+                        <xsl:value-of select="'info:eu-repo/semantics/restrictedAccess'"/>
+                    </xsl:when>
+                    <xsl:when test="$access-rights-datacite = 'REQUEST_PERMISSION'">
+                        <xsl:value-of select="'info:eu-repo/semantics/restrictedAccess'"/>
+                    </xsl:when>
+                    <xsl:when test="$access-rights-datacite = 'NO_ACCESS'">
+                        <xsl:value-of select="'info:eu-repo/semantics/closedAccess'"/>
+                    </xsl:when>
+                    <xsl:otherwise>
+                        <xsl:value-of select="."/>
+                    </xsl:otherwise>
+                </xsl:choose>
+            </xsl:element>
+        </xsl:if>
+
         <xsl:if test="$licenses">
             <xsl:for-each select="$licenses">
                 <xsl:choose>
-                    <xsl:when test=". = 'accept' and ../dcterms:accessRights = 'OPEN_ACCESS'">
+                    <xsl:when test=". = 'accept' and $access-rights-datacite = 'OPEN_ACCESS'">
```

Comment:
```suggestion
                    <xsl:when test=". = 'accept' and ../dcterms:accessRights = 'OPEN_ACCESS'">
```

Reply:
leave this as it was, we can only infer CC-0 if the accessCategory is OpenAccess and the 'accept' flag is set, because then it was chosen through the webui

Reply:
OK, done